### ISSUE TITLE
how to use Select2 Jquery in a php artisan preset react application

### ISSUE THREAD

**Issue by TOAO1** [jquery/jquery#4192]
I trying to integrate Laravel with Reactjs using php artisan preset react. In a normal create-react-app application, i would put Jquery cdn in the public/index.html like this:

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>

    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>`

And use it inside a react component like this: 

import React, { Component } from 'react';

import { Route, Link } from "react-router-dom";
import Carousel from './Carousel';
import Tabs from './Tabs';

import logo from '../../resources/netdi-logo.png'

class HeaderTop extends Component {
    constructor(props) {
        super(props);
        this.state = {
            Provinces: [],
        };
        this.select2TP = this.select2TP.bind(this);
        this.select2QH = this.select2QH.bind(this);
        this.select2PX = this.select2PX.bind(this);
    }

    select2QH(){
        window.$(document).ready(function() {
            window.$('#quan-huyen').select2();
        });
    }
    select2TP(){
        window.$(document).ready(function() {
            window.$('#thanh-pho').select2({
                width: 'resolve'
            });
        });
    }
    select2PX(){
        window.$(document).ready(function() {
            window.$('#phuong-xa').select2();
        });
    }

    componentDidMount(){
        this.select2PX();
        this.select2QH();
        this.select2TP();
    }

    render() {
        return (               
            .... content in here
        );
    }
}
    export default HeaderTop;

This work perfectly fine in create-react-app. However, it seem to be not the case in my laravel preset react application.

I try to use Jquery cdn in esources/view/App.blade.php:

`

<!DOCTYPE html>
    <html lang="{{ app()->getLocale() }}">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        
        <meta name="csrf-token" content="{{ csrf_token() }}">
        <title>Netdi Laravel-React</title>

        
        <script src="./js/popper.js"></script>

        <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>

        <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

        
        <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.css" />
        <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.min.js"></script>

        
        <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-bar-rating/1.2.1/themes/bars-reversed.min.css" />
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery-bar-rating/1.2.1/jquery.barrating.min.js"></script>

        
        <script src="https://maps.googleapis.com/maps/api/js?v=3.exp&sensor=false"></script>

        
        <link href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.6-rc.0/css/select2.min.css" rel="stylesheet" />
        <script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.6-rc.0/js/select2.min.js"></script>
        
        
        <link rel="stylesheet" href="./css/normalize.min.css" />
        <link rel="stylesheet" href="./css/bootstrap.min.css" />
        <link rel="stylesheet" href="./css/common.css" />
        <link rel="stylesheet" href="./css/home-page.css" />
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css" integrity="sha384-mzrmE5qonljUremFsqc01SB46JvROS7bZs3IO2EmfFsd15uHvIt+Y8vEf7N7fWAU" crossorigin="anonymous">

    </head>
    <body>
        <div id="app"></div>
        <div id="footer"></div>
        <script src="{{ asset('js/app.js') }}"></script>
    </body>
</html>`

And then, in assets/js/Example.js:

import React, { Component } from 'react';

// REACT ROUTER
import { Route, Link } from "react-router-dom";

// HEADERBODY'S COMPONENT
import Carousel from './Carousel'
import Tabs from './Tabs';

// RESOURCES
import logo from '../../resources/netdi-logo.png'

class HeaderTop extends Component {
    constructor(props) {
        super(props);
        this.state = {
                 Provinces: [],
        };
        this.select2TP = this.select2TP.bind(this);
        this.select2QH = this.select2QH.bind(this);
        this.select2PX = this.select2PX.bind(this);
    }

    select2QH(){
        window.$(document).ready(function() {
            window.$('#quan-huyen').select2();
        });
    }
    select2TP(){
        window.$(document).ready(function() {
            window.$('#thanh-pho').select2({
                width: 'resolve'
            });
        });
    }
    select2PX(){
        window.$(document).ready(function() {
            window.$('#phuong-xa').select2();
        });
    }

    componentDidMount(){
        this.select2PX();
        this.select2QH();
        this.select2TP();
    }

    render() {
        return (               
            .// a bunch of jsx content in here
        );
    }
}
export default HeaderTop;

And of course, i create another file that use ReactDOM to render this. When i run "npm run dev" it compiles successfully. But when i run "php artisan serve" it render nothing and return error: window.$(...).select2 is not a function. Is there a way to fix this?
